nuclear/module/NUClearModule.cmake

```diff
@@ -191,6 +191,14 @@ function(NUCLEAR_MODULE)
   # Put it in an IDE group for the module's directory
   set_target_properties(${module_target_name} PROPERTIES FOLDER ${module_path})
 
+  # ####################################################################################################################
+  # Warnings #
+  # ####################################################################################################################
+
+  include(CompilerWarnings)
+  # * Set the warnings with this function for the module
+  set_target_warnings(${module_target_name})
```

Comment:
Also apply to `test_module_target_name` target to apply the same scrutiny to tests. Also don't forget to apply to the tests for utility as well. Hopefully catch code isnt too nasty lol